.github/workflows/run_pipeline.yaml

```diff
@@ -0,0 +1,81 @@
+# Purpose:  Run the CI/CD pipeline that tests, packages, and publishes ScubaGear.
+# Note:  This pipeline is a work in progress.  At the moment, it is only doing linting, syntax checking, and unit testing.
+
+name: CI/CD Pipeline
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      # When set to true, it will run every step in the pipeline, regardless of
+      # what files have changed.
+      doEverything:
+        description: "Run every workflow in the pipeline."
+        required: false
+        type: boolean
+        default: true
+
+jobs:
+  test-files:
+    name: Test for Changes
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Check for Changes
+        uses: dorny/paths-filter@v3
+        id: file-changes
+        with:
+          base: ${{ github.ref }}
+          filters: |
+            yaml-files:
+              - added|modified: "**.yml"
+              - added|modified: "**.yaml"
+            powershell-files:
+              - added|modified: "**.ps1"
+              - added|modified: "**.psm1"
+              - added|modified: "**.psd1"
+              - added|modified: "**.pssc"
+              - added|modified: "**.psrc"
+              - added|modified: "**.ps1xml"
+              - added|modified: "**.cdxml"
+            markdown-files:
+              - added|modified: "PowerShell/ScubaGear/baselines/*.md"
+            rego-files:
+              - added|modified: "**.rego"
+    outputs:
+      yaml-changes: ${{ steps.file-changes.outputs.yaml-files || inputs.doEverything }}
```

Comment:
Is my assumption that on a pull/push request the inputs.doEverything resolves to false?

Reply:
Yes, I think so, but I won't know for sure until it's pushed to main.  Trunk-based development for the win!